services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersValidateSimpleFlowSpec.groovy

```diff
@@ -12,69 +12,64 @@ import groovy.transform.Memoized
-        def burstSizeDstSwitch = switchHelper.getExpectedBurst(dstSwitch.dpId, flow.maximumBandwidth)
-
-        createdMetersIdSrcSwitch.size() == 1
-        createdMetersIdDstSwitch.size() == 1
+        then: "One meter is automatically created on the src and dst switches"
+        def srcSwitchCreatedMetersId = getCreatedMetersId(srcSwitch.dpId, srcSwitchDefaultMeters)
+        def dstSwitchCreatedMetersId = getCreatedMetersId(dstSwitch.dpId, dstSwitchDefaultMeters)
+        srcSwitchCreatedMetersId.size() == 1
+        dstSwitchCreatedMetersId.size() == 1
 
         and: "The correct info is stored in the validate meter response"
-        def meterValidateInfoSrcSwitch = northbound.validateMeters(srcSwitch.dpId)
-        def meterValidateInfoDstSwitch = northbound.validateMeters(dstSwitch.dpId)
-        def createdCookieSrcSwitch = getCreatedCookies(srcSwitch.dpId, createdMetersIdSrcSwitch)
-        def createdCookieDstSwitch = getCreatedCookies(dstSwitch.dpId, createdMetersIdDstSwitch)
+        def srcSwitchValidateInfo = northbound.validateMeters(srcSwitch.dpId)
+        def dstSwitchValidateInfo = northbound.validateMeters(dstSwitch.dpId)
+        def srcSwitchCreatedCookie = getCreatedCookies(srcSwitch.dpId, srcSwitchCreatedMetersId)
+        def dstSwitchCreatedCookie = getCreatedCookies(dstSwitch.dpId, dstSwitchCreatedMetersId)
 
-        def createdCookies = createdCookieSrcSwitch + createdCookieDstSwitch
+        srcSwitchValidateInfo.meters.proper*.meterId.containsAll(srcSwitchCreatedMetersId)
+        dstSwitchValidateInfo.meters.proper*.meterId.containsAll(dstSwitchCreatedMetersId)
 
-        meterValidateInfoSrcSwitch.meters.proper.collect { it.meterId }.containsAll(createdMetersIdSrcSwitch)
-        meterValidateInfoDstSwitch.meters.proper.collect { it.meterId }.containsAll(createdMetersIdDstSwitch)
-
-        meterValidateInfoSrcSwitch.meters.proper.collect { it.cookie }.containsAll(createdCookieSrcSwitch)
-        meterValidateInfoDstSwitch.meters.proper.collect { it.cookie }.containsAll(createdCookieDstSwitch)
+        srcSwitchValidateInfo.meters.proper*.cookie.containsAll(srcSwitchCreatedCookie)
+        dstSwitchValidateInfo.meters.proper*.cookie.containsAll(dstSwitchCreatedCookie)
 
-        [meterValidateInfoSrcSwitch, meterValidateInfoDstSwitch].each {
+        [srcSwitchValidateInfo, dstSwitchValidateInfo].each {
             it.meters.proper.each {
                 assert it.rate == flow.maximumBandwidth
                 assert it.flowId == flow.id
             }
-        }
-
-        meterValidateInfoSrcSwitch.meters.proper.burstSize[0] == burstSizeSrcSwitch
-        meterValidateInfoDstSwitch.meters.proper.burstSize[0] == burstSizeDstSwitch
-
-        [meterValidateInfoSrcSwitch, meterValidateInfoDstSwitch].each {
             it.meters.proper.every {
-                ["KBPS", "BURST", "STATS"].containsAll(it.flags)
+                assert ["KBPS", "BURST", "STATS"].containsAll(it.flags)
```

Comment:
should be inside `it.meters.proper.each {` on L52. Why loop twice?

Reply:
Fixed